opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/TopologyRestService.java

```diff
@@ -0,0 +1,142 @@
+import org.springframework.stereotype.Component;
+
+@Component
+@Path("topology")
+public class TopologyRestService {
+/*
+    private final static String ID = "id";
+    private final static String DESCRIPTION = "description";
+    private final static String IP_ADDRESS = "ipAddr";
+    private final static String LOCKED = "locked";
+    private final static String FOREIGN_SOURCE = "foreignSource";
+    private final static String FOREIGN_ID = "foreignID";
+    private final static String SELECTED = "selected";
+    private final static String STYLE_NAME = "styleName";
+    private final static String X = "x";
+    private final static String Y = "y";
+    private final static String PREFERRED_LAYOUT = "preferred-layout";
+    private final static String FOCUS_STRATEGY = "focus-strategy";
+    private final static String FOCUS_IDS = "focus-ids";
+    private final static String SEMANTIC_ZOOM_LEVEL = "semantic-zoom-level";
+    private final static String VERTEX_STATUS_PROVIDER = "vertex-status-provider";
+    private final static String LEVEL = "level";
+    private final static String EDGE_PATH_OFFSET = "edge-path-offset";
+    private final static String BREADCRUMB_STRATEGY = "breadcrumb-strategy";
+  */  
+    private final static String TOPOLOGY_NAMESPACE_PREFIX="Topology::";
+    private final static String NAMESPACE = "namespace";
+    private final static String ICON_KEY = "iconKey";
+    private final static String LABEL = "label";
+    private final static String NODE_ID = "nodeID";
+    private final static String TOOLTIP_TEXT = "tooltipText";
+//FIXME support port
+//    private final static String SOURCE_PORT= "sourceport";
+//    private final static String TARGET_PORT= "targetport";
+    private final static String SOURCE_IFINDEX= "sourceifindex";
+    private final static String TARGET_IFINDEX= "targetifindex";
+    
+    private TopologyDao m_topologyDao;
+
+
+    @GET
+    @Path("{supported-protocol}")
+    public Response getGraph(@PathParam("supported-protocol") String supportedprotocol) throws InvalidGraphException {
+        if (supportedprotocol.equals(Topology.ProtocolSupported.CDP.name())) {
+            return Response.ok(getGraph(ProtocolSupported.CDP)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.LLDP.name())) {
+            return Response.ok(getGraph(ProtocolSupported.LLDP)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.BRIDGE.name())) {
+            return Response.ok(getGraph(ProtocolSupported.BRIDGE)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.OSPF.name())) {
+            return Response.ok(getGraph(ProtocolSupported.OSPF)).build();
+        }
+        if (supportedprotocol.equals(Topology.ProtocolSupported.ISIS.name())) {
+            return Response.ok(getGraph(ProtocolSupported.ISIS)).build();
+        }
+        return Response
```

Comment:
If not supported, we should return `Response.Status.Not_FOUND`. The `SERVICE_UNAVAILABLE` does not work in this case.

Reply:
OK